### PR TITLE
Return 405 correctly even if version is using as header and wrong request method

### PR DESCRIPTION
**Pull request by namusyaka**
This reverts commit a0166e77468093601975e7005ec2e2a46e3305ae.

I'll try to fix this problem drastically.

### CODE REVIEW

**Comment by namusyaka** [comment]
@dblock Could you take a look this pull request?
I think this approach can fix potential bugs and issues.
If ok, we can release 0.16.2 version ;)

File: lib/grape/endpoint.rb [hunk 1]
```diff
@@ -231,7 +231,7 @@ def run
 
         run_filters before_validations, :before_validation
 
-        run_validators validations, request unless @method_not_allowed
+        run_validators validations, request unless env[Grape::Env::GRAPE_METHOD_NOT_ALLOWED]
```
Comment (dblock) [inline lib/grape/endpoint.rb]: I see it now :) Better.

Comment (dblock) [inline lib/grape/endpoint.rb]: You might want to audit the use if `@variable` elsewhere, we have at least an `@app_response` in a formatter middleware, `@named_params` in a DSL helper and `@versions` in routing.

**Comment by dblock** [comment]
This worked everywhere where I saw problems, feel free to cut 0.16.2, which is https://github.com/ruby-grape/grape/issues/1360. I would review the use of `@...` first though.

**Comment by namusyaka** [comment]
The `@method_not_allowed` variable usage is wrong. It returns `true` eternally if method_not_allowed is set  even once.
Other `@~` vars looks like no problem.

**Comment by dblock** [comment]
I should have caught `@` in the code reviews too, my bad.

**Comment by namusyaka** [comment]
I've mistaken the coding. Sorry for the bad patch. Btw, we can release the v.0.16.2 now. I'll prepare it soon.

**Pull request merged**